dask_kubernetes/operator/daskcluster.py

```diff
@@ -237,9 +245,15 @@ async def daskworkergroup_update(spec, name, namespace, logger, **kwargs):
             )
         logger.info(f"Scaled worker group {name} up to {spec['replicas']} workers.")
     if workers_needed < 0:
-        for i in range(current_workers, desired_workers, -1):
+        nest_asyncio.apply()
```

Comment:
I am curious why this needs to be done?

Reply:
The library [`nest-asyncio`](https://github.com/erdewit/nest_asyncio) allows for nested event loops in `asyncio`. I wondered if you another of doing this without it?

Reply:
I guess I'm asking why we need to nest the event loop. Everything should just be able to use the default one. Kopf suppoers asyncio and you are already inside an async function.